pilot/pkg/bootstrap/configcontroller.go

```diff
@@ -225,7 +225,11 @@ func (s *Server) initK8SConfigStore(args *PilotArgs) error {
 			leaderelection.
 				NewLeaderElection(args.Namespace, args.PodName, leaderelection.IORController, args.Revision, s.kubeClient).
 				AddRunFunction(func(leaderStop <-chan struct{}) {
-					ior.Run(s.kubeClient, configController, leaderStop)
+					controller := ior.New(s.kubeClient, configController)
+
+					s.kubeClient.RunAndWaitRouteInformer(leaderStop)
+
+					controller.Run(leaderStop)
```

Comment:
Could not it be simple like this?
```suggestion
                                        s.kubeClient.RunAndWait(stop)
					ior.Run(s.kubeClient, configController, leaderStop)
```
Then you could revert all other changes from this PR.

Reply:
Ok, now I see that it couldn't work, because `RunAndWait()` must be called after `.Informer()`.